analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -81,6 +81,14 @@ private static void ReportIssue(ReportingContext reportingContext, bool? isTestP
                 return;
             }
 
+            if (reportingContext.Compilation is { } compilation
+                && reportingContext.Diagnostic.Location.Kind == LocationKind.SourceFile
+                && !compilation.ContainsSyntaxTree(reportingContext.Diagnostic.Location.SourceTree.GetRoot()?.SyntaxTree))
```

Comment:
Could be rewritten as such (given that `GetRoot()?.SyntaxTree` is superfluous).
```suggestion
            if (reportingContext is
                {
                    Compilation: { } compilation,
                    Diagnostic.Location:
                    {
                        Kind: LocationKind.SourceFile,
                        SourceTree: { } syntaxTree
                    }
                }
                && !compilation.ContainsSyntaxTree(syntaxTree))
```